### DISCUSSION TITLE
Console blocking output

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1524] [Ideas]
Couldn't it be possibly done, that having the console open would block inputs from being registered in the game, but only to the console? (eg. other menus won't open)

**Comment by kagikn** [comment]
The console actually does disable controls except for camera look ones (unless some scripts call "enable control" natives because "disable" and "enable" control natives access to the same global data), and SHVDN doesn't send keyboard messages to scripts at all when the console is open. Scripts can do something on tick even when the console is open though, so they can react to control inputs. There's a native that can return true if the specified control is pressed regardless of it being enabled even. In reality, SHVDN can't really block only inputs so script can't use them at all, not entire tick functions (in simple ways).

**Reply by Michael21107** [reply]
Yeah, I was even thinking of the blocking of inputs from RPH plugins, so I guess that's absolutely impossible, right?

**Reply by kagikn** [reply]
The game doesn't send events such as control pressing events to scripts (including RPH plugins) but scripts themselves can do their stuff such as detecting whether certain controls are pressed only when being ticked in Rockstar's scripting engines in their games. But yes, SHVDN can't block scripts other than SHVDN ones or RPH plugins reacting to control inputs at all, without stopping the entire functions of the scripts.

**Reply by Michael21107** [reply]
gotcha, thanks a lot
